src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -6251,5 +6251,191 @@ .maxstack  2
   IL_0003:  ret
 }");
         }
+
+        [Fact]
+        public void ConstantFolding()
+        {
+            constantFolding("nint", "const nint A = -2147483648;", "+A", "-2147483648");
```

Comment:
Can you please add some comments clarifying the choices of numbers? I assume that these are max values of something, but I don't have the max values of various integer types memorized.